packages/block-editor/src/components/use-block-display-information/index.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as blocksStore } from '@wordpress/blocks';
+
+/** @typedef {import('@wordpress/blocks').WPIcon} WPIcon */
+
+/**
+ * Contains basic block's information for display reasons.
+ *
+ * @typedef {Object} WPBlockDisplayInformation
+ *
+ * @property {string} title Human-readable block type label.
+ * @property {WPIcon} icon Block type icon.
+ * @property {string} description A detailed block type description.
+ */
+
+/**
+ * Hook used to try to find a matching block variation and return
+ * the appropriate information for display reasons. In order to
+ * to try to find a match we need to things:
+ * 1. Block's client id to extract it's current attributes.
+ * 2. A block variation should have set `isActive` prop to a proper function.
+ *
+ * If for any reason a block variaton match cannot be found,
+ * the returned information come from the Block Type.
+ * If no blockType is found with the provided clientId, returns null.
+ *
+ * @param {string} clientId Block's client id.
+ * @return {?WPBlockDisplayInformation} Block's display information.
+ *
+ */
+
+// TODO write tests
+export default function useBlockDisplayInformation( clientId ) {
+	const { attributes, blockType, variations } = useSelect(
```

Comment:
Is it possible to move all logic to `useSelect`?

```js
export default function useBlockDisplayInformation( clientId ) {
    return useSelect(
        // all logic here
    );
}
```

Reply:
moved here: https://github.com/WordPress/gutenberg/pull/27469/commits/a42d2b22288484aebf42b6dea8ac30656bbab7ed